packages/perseus/src/widgets/interactive-graphs/graphs/components/movable-point.tsx

```diff
@@ -16,18 +16,20 @@ type Props = {
     onMove: (newPoint: vec.Vector2) => unknown;
     color?: string;
     cursor?: CSSCursor | undefined;
+    snapTo?: "grid" | "angles" | "sides";
 };
 
 export const StyledMovablePoint = (props: Props) => {
     const {snapStep} = useGraphConfig();
     const elementRef = useRef<SVGGElement>(null);
-    const {point, onMove, cursor, color = WBColor.blue} = props;
-
+    const {point, onMove, cursor, color = WBColor.blue, snapTo} = props;
+    const snapToValue = snapTo ?? "grid";
     const {dragging} = useMovable({
         gestureTarget: elementRef,
         point,
         onMove,
-        constrain: (p) => snap(snapStep, p),
+        constrain: (p) =>
+            ["angles", "sides"].includes(snapToValue) ? p : snap(snapStep, p),
```

Comment:
The tests I added were meant to test this and the similar constrain update in the polygon file. They still pass when this is returned to the original constrain. Would love suggestions on how to test this.

Reply:
I think that in order to test this logic, you need a test where:

- angle snapping is on
- you try to move a point to a certain (x, y)
- angle snapping causes it to end up at a point different from (x, y)

So, e.g. in this test:

```ts
const updated = interactiveGraphReducer(
    state,
    movePoint(0, [3.130341262927298, -2.932437813845241]),
);

invariant(updated.type === "polygon");
expect(updated.coords[0]).toEqual([
    3.130341262927298, -2.932437813845241,
]);
```

The coords passed to `movePoint` need to be different from the coords in the expectation.

This is the kind of test where I'd run it once to find out what the resulting point coords are, and then just paste those into the expectation. The correctness of the resulting test will be very non-obvious, but at least it will guard against regressions.

Reply:
I can change it to be that direction for sure. Unfortunately, when I change the constrain on `StyledMovablePoint` back to just `snap(snapStep,p)`, the polygon points start snapping to the grid, and the test still passes.
<img width="986" alt="image" src="https://github.com/Khan/perseus/assets/60857422/b38b1e62-a6ba-4f05-9853-e6eb0592135c">

Reply:
oh, my bad... I misread the code. The reducer tests don't invoke this code at all, since it's in the point component, not the reducer. To test the `constrain`, you'd have to write a React Testing Library test.

I actually think you *can* revert the constrain to `snap(snapStep, p)` and everything will work fine. The reducer will make sure the points are snapped correctly.

This `constrain` is only needed for keyboard support (it makes sure that pressing an arrow key always moves the point by a single snap step). Since we've decided not to pursue keyboard support for angle snapping, I think we can leave it as-is.

Reply:
When I revert the constrain, it starts snapping to the grid, but the tests don't catch it. Here's a video showing that, though it does use the flipbook. I'll start working on a RTL test for this tomorrow unless you still don't think it's needed. :)

https://github.com/Khan/perseus/assets/60857422/93db6f30-c7ce-491a-a250-973571c12a17

Reply:
Ah, if we can't use keyboard to move a point with angle snapping, then I don't think I can test using RTL, right? I'll just move forward with this and we can change things or add tests later if we need to :)